iothub/device/devdoc/API Design.md

```diff
@@ -187,41 +395,89 @@ public string AckDescription { get; set; }
+<details>
+<summary>TelemetryConvention</summary>
 
-/// <summary>
-/// Send telemetry using the specified message.
-/// </summary>
-/// <remarks>
-/// Use this method when you need to define custom properties for the message.
-/// </remarks>
-/// <param name="telemetryMessage">The custom implemented telemetry message</param>
-/// <param name="componentName">The component name this telemetry belongs to.</param>
-/// <param name="cancellationToken">A cancellation token.</param>
-public Task SendTelemetryAsync(Message telemetryMessage, string componentName = default, CancellationToken cancellationToken = default);
+```csharp
+public class TelemetryConvention
+{
+    /// <summary>
+    /// The content type for a plug and play compatible telemetry message.
+    /// </summary>
+    private const string ApplicationJson = "application/json";
+
+    /// <summary>
+    ///
+    /// </summary>
+    public static readonly TelemetryConvention Instance = new TelemetryConvention();
+
+    /// <summary>
+    ///
+    /// </summary>
+    public Encoding ContentEncoding { get; set; } = Encoding.UTF8;
+
+    /// <summary>
+    ///
+    /// </summary>
+    public string ContentType { get; set; } = ApplicationJson;
+
+    /// <summary>
+    /// Format a plug and play compatible telemetry message payload.
+    /// </summary>
+    /// <param name="telemetryName">The name of the telemetry, as defined in the DTDL interface. Must be 64 characters or less. For more details see
+    /// <see href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#telemetry"/>.</param>
+    /// <param name="telemetryValue">The unserialized telemetry payload, in the format defined in the DTDL interface.</param>
+    /// <returns>A plug and play compatible telemetry message payload, which can be sent to IoT Hub.</returns>
+    public static IDictionary<string, object> FormatTelemetryPayload(string telemetryName, object telemetryValue)
+    {
+        return new Dictionary<string, object> { { telemetryName, telemetryValue } };
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="objectToSerialize"></param>
+    /// <returns></returns>
+    public virtual string SerializeToString(object objectToSerialize)
+    {
+        return JsonConvert.SerializeObject(objectToSerialize);
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="contentPayload"></param>
+    /// <returns></returns>
+    public virtual byte[] EncodeStringToByteArray(string contentPayload)
+    {
+        return ContentEncoding.GetBytes(contentPayload);
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="objectToSendWithConvention"></param>
+    /// <returns></returns>
+    public virtual byte[] GetObjectBytes(object objectToSendWithConvention)
+    {
+        return EncodeStringToByteArray(SerializeToString(objectToSendWithConvention));
+    }
+}
 ```
+</details>
 
 ### Commands
 
```

Comment:
SubscribeForCommands

Reply:
single global command callback

Reply:
should we have a convention for commands as well (just for symmetry) - deserialization